### PR TITLE
1040 fix upload binary and download thru CW

### PR DESCRIPTION
**Pull request by leite08**
Ticket: https://github.com/metriport/metriport-internal/issues/1040

### Dependencies

- Upstream: https://github.com/metriport/metriport-internal/pull/1463
- Downstream: none

### Description

- only return docrefs uploaded by cx
- update docref attachment URL before returning to CW's DQ

### Testing

- Local
  - none
- Staging
  - [ ] ask to upload a file but don't - it doesn't get returned to CW
  - [ ] upload file using `POST /document/upload` and download from another org/cx
- Sandbox
  - none
- Production
  - none

### Release Plan

- nothing special

### CODE REVIEW

File: packages/api/src/external/commonwell/proxy/process-outbound.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    const payload = JSON.parse(payloadString);
+    // Filter out CW data while we don't manage to do it with FHIR query
+    if (payload.entry) {
+      payload.entry = filterDataForCW(payload.entry);
+      payload.entry = adjustAttachmentURLs(payload.entry);
+      payload.total = payload.entry?.length != null ? payload.entry.length : undefined;
+    }
+    const response = JSON.stringify(payload);
+    log(`Responding to CW (cxPatientId ${cxPatientId}): ${response}`);
+    return response;
```
[the remark this answers is not in the document]
Comment (leite08) [reply packages/api/src/external/commonwell/proxy/process-outbound.ts:39]: yeah, I can ticket that.

File: packages/api/src/external/commonwell/proxy/cw-fhir-proxy.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+);
+
+const router = fhirServerUrl
+  ? proxy(fhirServerUrl, {
+      proxyReqPathResolver: proxyRequest,
+      userResDecorator: processResponse,
+    })
+  : dummyRouter;
+
+export default router;
```
[the remark this answers is not in the document]
Comment (leite08) [reply packages/api/src/external/commonwell/proxy/cw-fhir-proxy.ts:24]: It's already doing this, not as clean though.

**Review by RamilGaripov — APPROVED** [review]
LGTM 🚀

**Review by Goncharo — APPROVED** [review]
:shipit:

File: packages/api/src/external/commonwell/proxy/process-outbound.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (leite08) [reply packages/api/src/external/commonwell/proxy/process-outbound.ts:39]: https://github.com/metriport/metriport-internal/issues/1465

**Pull request merged**